htdocs/class/xoopsform/renderer/XoopsFormRendererTailwind.php

```
@@ -0,0 +1,738 @@
+    {
+        if (isset($GLOBALS['xoTheme'])) {
+            $GLOBALS['xoTheme']->addScript('include/spectrum.js');
+            $GLOBALS['xoTheme']->addStylesheet('include/spectrum.css');
+        } else {
+            echo '<script type="text/javascript" src="' . XOOPS_URL . '/include/spectrum.js"></script>';
+            echo '<link rel="stylesheet" type="text/css" href="' . XOOPS_URL . '/include/spectrum.css">';
+        }
+        return '<input class="input input-bordered w-24 h-10 p-1" type="color" name="' . $element->getName()
+            . '" title="' . $element->getTitle() . '" id="' . $element->getName()
+            . '" size="7" maxlength="7" value="' . $element->getValue() . '"' . $element->getExtra() . '>';
+    }
```

PHPMD (2.15.0)

[error] renderFormColorPicker accesses the super-global variable $GLOBALS. (undefined)

(Superglobals)